### ISSUE TITLE
Tx Serialization Mismatch

### ISSUE THREAD

**Issue by nemo83** [vacuumlabs/cardano-hw-cli#187]
Hello,

I'm not sure where to start from, but I'm getting the following error when signing a Stake Pool Certificate.

```
Adding the owner witness from a local Hardware-Wallet key 'ledgerowner', continue? [Y/n] Y
Cardano App Version 7.1.4 (HW-Cli Version 1.16.0-rc.1) found on your Ledger device!
Please approve the action on your Hardware-Wallet (abort with CTRL+C) ... Error: The requested operation failed. Check that your Ledger device is connected, unlocked and with Cardano app running.
Details: Error: The requested operation failed. Check that your Ledger device is connected, unlocked and with Cardano app running.
Details: Error: Tx serialization mismatch
```

Some info:
* Using Nano X on 2.2.4
* Cardano app 7.1.4
* cardano-cli 9.2.1.0 - darwin-x86_64 - ghc-8.10 \ git rev efd560070aaf042d1eb4680ae37fc607c7742319
* Cardano HW CLI Tool version 1.16.0-rc.1 \ Commit hash: e7225d91e0ddc6152ad973ba803d6c016cb14d7e

I'm using Martin's ATADA scripts to sign all-the-things.

Any idea?

Thanks in advance

**Comment by gitmachtl** [comment]
Hey, what kind of transaction is this? Stakepool registration update?

**Comment by nemo83** [comment]
`05c_regStakepoolCert.sh pool_name owner.payment.addr`

**Comment by gitmachtl** [comment]
Did you use a CLI wallet to pay for the transaction fee? Or did you try to use the funds from the ledger itself to pay for the transaction? If so, can you try to use some little funds from a CLI address instead and repeat the transaction?

**Comment by nemo83** [comment]
Hey, thanks for the fast feedback!

So what I'm trying to do here is to:
1. add a cli owner wallet
2. change rewards address, from ledger to cli

This tx was paid w/ the new owner cli small wallet. Should I use a complete independent small wallet?

EDIT: just tried with a small wallet (non-owner) and same issue.

**Comment by gitmachtl** [comment]
No that should not have an impact. So currently you have a ledger registered as the pool owner and you want to add a cli owner, ok.

Will try to reproduce this when i am back home on a testnet.

**Comment by nemo83** [comment]
> No that should not have an impact. So currently you have a ledger registered as the pool owner and you want to add a cli owner, ok.
> 
> Will try to reproduce this when i am back home on a testnet.

Thank you sir! Just for completeness sake, the ledgerowner, is a full HW account (non hybrid).

**Comment by gitmachtl** [comment]
> > No that should not have an impact. So currently you have a ledger registered as the pool owner and you want to add a cli owner, ok.
> > Will try to reproduce this when i am back home on a testnet.
> 
> Thank you sir! Just for completeness sake, the ledgerowner, is a full HW account (non hybrid).

one question, did you run 05a again before running 05c? just to make sure the right certificate was generated and its not an old one laying around.

**Comment by nemo83** [comment]
> > > No that should not have an impact. So currently you have a ledger registered as the pool owner and you want to add a cli owner, ok.
> > > Will try to reproduce this when i am back home on a testnet.
> > 
> > 
> > Thank you sir! Just for completeness sake, the ledgerowner, is a full HW account (non hybrid).
> 
> one question, did you run 05a again before running 05c? just to make sure the right certificate was generated and its not an old one laying around.

Yeah, just tried again. Also, I'm using macosx

**Comment by gitmachtl** [comment]
ah ok, hmm. can only test on regular linux. what bash version do you run?

**Comment by nemo83** [comment]
```/usr/local/bin/bash --version
GNU bash, version 5.2.26(1)-release (x86_64-apple-darwin23.2.0)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.```

**Comment by gitmachtl** [comment]
lets see if i stumple across an issue when i test it.

**Comment by gitmachtl** [comment]
so far, i tested with 7.1.3 on a nano s+ and that was fine:

```
Adding the owner witness from a local Hardware-Wallet key 'ledger-purple', continue? [Y/n]
Cardano App Version 7.1.3 (HW-Cli Version 1.16.0-rc.1) found on your Ledger device!
Please approve the action on your Hardware-Wallet (abort with CTRL+C) ... DONE
```

updated to cardano-app 7.1.4 and retested:
```
Adding the owner witness from a local Hardware-Wallet key 'ledger-red3', continue? [Y/n]
Cardano App Version 7.1.4 (HW-Cli Version 1.16.0-rc.1) found on your Ledger device!
Please approve the action on your Hardware-Wallet (abort with CTRL+C) ... DONE
```

i than repeated the same tests with a Nano-X hw wallet, same result. 

i tested with only the hw wallet as a pool owner, as a pool owner and rewards account, with an additional cli owner and with a cli rewards account. i even tested it with the pool keys on the ledger. all worked fine.

can you do the following test, your ledger name is `ledgerowner`, can you generate a new stakeaddress/paymentaddress from the same hw wallet like `./03a_genStakingPaymentAddr.sh ledgertest hw` and check if the address and the vkey from that `ledgertest` is the same as your `ledgerowner` ?

**Comment by nemo83** [comment]
Right, I'm extremely sorry, but it actually all worked fine.

I was using a ledger S to manage the pool, which after different upgrades, it could not be used anymore to sign pool certificates. So I switched to Nano X and restored a mnemonic. Never realising it was a different wallet.

When comparing the keys as you suggested I found out it was different. Once I restored the correct mnemonic it worked right away!

Sorry for wasting your time Martin 😭
